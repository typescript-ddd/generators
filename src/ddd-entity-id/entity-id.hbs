import { EntityId{{#if isUuid}}, Uuid{{/if}} } from "@typescript-ddd/core";

/**
 * Represents the unique identifier for {{noun entityName}}.
 */
export class {{pascal entityName}}Id extends EntityId {
  /**
   * Initializes a new instance of {{noun entityName}} identifier.
   * @param {{lbrace}}{{idType}}{{rbrace}} value - The identity value.
   */
  constructor(value: {{idType}}) {
    super(value);
  }

  /**
   * Creates a new instance of {{noun entityName}} identifier.
   * @returns {{lbrace}}{{entityName}}{{rbrace}}
   */
  public static create(): {{pascal entityName}}Id {
    {{#if isUuid}}
    return new {{pascal entityName}}Id(Uuid.create());
    {{else}}
    return new {{pascal entityName}}Id({{valueCreator}});
    {{/if}}
  }

  /**
   * Creates a new instance of {{noun entityName}} identifier.
   * @param {{lbrace}}{{idType}}{{rbrace}} value - The identity value.
   * @returns {{lbrace}}{{pascal entityName}}Id{{rbrace}} - A new instance of {{noun entityName}} identifier.
   */
  public static from(value: {{idType}}): {{pascal entityName}}Id {
    return new {{pascal entityName}}Id(value);
  }
}