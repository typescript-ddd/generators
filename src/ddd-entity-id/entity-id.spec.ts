import { EntityIdTemplateData } from "./entity-id.types";
import { generateEntityId } from "./entity-id.generator";

describe("EntityId", () => {
    it("should render entity id template", () => {
        // Arrange
        const data: EntityIdTemplateData = {
            entityName: "User",
            idType: "string",
            valueCreator: "Uuid"
        };

        const result = generateEntityId(data);
        expect(result).toMatchSnapshot();
        
    });
})