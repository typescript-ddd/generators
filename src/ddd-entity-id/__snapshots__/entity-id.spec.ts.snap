// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityId should render entity id template 1`] = `
"import { EntityId } from "@typescript-ddd/core";

/**
 * Represents the unique identifier for a user.
 */
export class UserId extends EntityId {
  /**
   * Initializes a new instance of a user identifier.
   * @param {string} value - The identity value.
   */
  constructor(value: string) {
    super(value);
  }

  /**
   * Creates a new instance of a user identifier.
   * @returns {User}
   */
  public static create(): UserId {
    return new UserId(Uuid);
  }

  /**
   * Creates a new instance of a user identifier.
   * @param {string} value - The identity value.
   * @returns {UserId} - A new instance of a user identifier.
   */
  public static from(value: string): UserId {
    return new UserId(value);
  }
}"
`;
